src/sourabh/scannerExamples/ShopScanner.java

```diff
@@ -0,0 +1,175 @@
+/*1. Create a system (Shop) which maintains each item's quantity.
+
+a. Initial stocks of Maggie: 50 Maggie packets 
+b. Initial stocks of Dosa : 43 packets 
+c. Initial stocks of pouches : 39 oil packets 
+d. Initial stocks of panipuri : 43 packets  
+e. Initial stocks of masala�s : 73 packets.
+
+Expectation:
+1) Only one Customer came for shopping, he picked different items with different quantities,
+print "running out of stock" message if a particular item is sold out otherwise deduct requested 
+quantity from total quantity. Partial selling is allowed.  
+2) Create a method which shows only those item which is "Out of stock".
+3) Create a method which shows only those item which is "Available in stock" & respective quantities.*/
+
+package sourabh.scannerExamples;
+
+import java.util.Scanner;
+
+public class ShopScanner {
+	int maggieQty=0;
+	int dosaQty=0;
+	int oilPacketsQty=0;
+	int panipuriQty=0;
+	int masalaQty=0;
+	void setQuantities(int maggieQtyini, int dosaQtyini, int oilPacketsQtyini,int panipuriQtyini, int masalaQtyini) {
+		maggieQty= maggieQtyini; 
+		dosaQty= dosaQtyini;
+		oilPacketsQty= oilPacketsQtyini;
+		panipuriQty= panipuriQtyini;
+		masalaQty= masalaQtyini;
+	}
+	void customerShopping(int maggieQtyReq, int dosaQtyReq, int oilPacketsQtyReq, int panipuriQtyReq, int masalaQtyReq) {
+		//Maggie
+		if(maggieQtyReq>maggieQty) {
+			System.out.println("Maggie running out of stock, you can purchase only "+maggieQty+" Maggie packets.");
+			int maggieQtyNew= maggieQtyReq- maggieQty;
+			System.out.println("We can provide you only "+maggieQty+" Maggie packets remaining "+maggieQtyNew+" you can buy later.");
+			maggieQty=0;
```

Comment:
If maggieQtyReq is more than total, then it doesn't mean we set total quantity to 0. right? @SourabhSaha93